src/main/kotlin/com/terraformation/backend/documentproducer/model/ExistingDocumentModel.kt

```diff
@@ -19,6 +22,7 @@ data class ExistingDocumentModel(
     val name: String,
     val ownedBy: UserId,
     val projectId: ProjectId,
+    val projectName: String? = null,
```

Comment:
Is it ever actually valid for this to be null, or is this just because the `DocumentsRow` constructor doesn't have a value for the project name?

If that's the only reason, maybe it would make sense to have the `DocumentsRow` constructor take the project name as a second argument, or alternately, replace all the uses of that constructor with the new `of()` function and make sure they pass in a record that includes the project name.

Reply:
It is only really used by the "list" calls, so maybe I should just make a separate "existing model". I looked at all the usages, and decided against refactoring everything using the model since it is used in a few different places. 

To answer your question, no, it technically isn't valid for it to be null, but based on where it is being called from, it may be null because it consumers don't care about it. Saying that out loud, perhaps it should be changed up. I have a subsequent PR that adds yet another field to this payload, which, again, is only there under certain circumstances. It seems that a new, slightly different model would make sense here.

Reply:
I noticed there was another ticket that needs the project name, for the "get single document" call. So I ended up replacing all uses of the Row constructor with the Record `of()` function. This is done in #2242.